cadc-sia/src/main/java/ca/nrc/cadc/sia2/impl/ServiceAvailability.java

```diff
@@ -155,17 +157,23 @@ public void setState(String string) {
      * otherwise attempt to use it as a URI and look the URL up in the Registry.
      *
      * @return URL Base URL of the TAP service to use.
+     *
      * @throws MalformedURLException If a URL cannot be created from the specified string.
      */
     public static URL getTapBaseURL() throws MalformedURLException {
         RegistryClient regClient = new RegistryClient();
         URI configuredTapURI = URI.create(ServiceAvailability.getTapURI());
 
         AuthMethod am = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
-        URL serviceURL = regClient.getServiceURL(configuredTapURI, Standards.TAP_10,
-                (am == null) ? AuthMethod.ANON : am);
+        try {
 
-        return (serviceURL == null) ? configuredTapURI.toURL() : serviceURL;
+            // Attempt to load the URI as a resource URI from the Registry.
+            return regClient.getServiceURL(configuredTapURI, Standards.TAP_10,
+                                           (am == null) ? AuthMethod.ANON : am);
+        } catch (IllegalArgumentException iae) {
+            // Fallback and assume the URI is an absolute one.
+            return configuredTapURI.toURL();
```

Comment:
IllegalArgumentException is only thrown when one of the params is null... you could check those in this code instead then you wouldn't be relying on the exception to make the decision.

Reply:
The `IllegalArgumentException` is thrown from the `getAccessURL` method unfortunately, and so needs to be dealt with.  That may be a bug.

Reply:
That IllegalArgumentException is at a minimum not thrown for the right reason. The reason it should be thrown is that the argument resourceID is not an ivo-scheme URI and it could thrown a ResourceNotFoundException or return null if the service is not "registered". 

The code here should not rely on that behaviour or use exception for logic. You should just check that the URI scheme is ivo before calling regClient and try toURL if it is not.